### PR TITLE
Version with setuptools_scm as in pyuvdata

### LINKED ISSUE #271: change how we handle versioning to use setuptools_scm as we do in pyuvdata
This change will mean that all versioning changes are handled by setting tags on the repo.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Use setuptools_scm to handle versioning, as we do in pyuvdata.

## Motivation and Context

closes #271

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [x] Build or continuous integration change

## Checklist:

Build or continuous integration change checklist:
- [x] If required or optional dependencies have changed (including version numbers), I have updated the readme to reflect this.
- [ ] If this is a new CI setup, I have added the associated badge to the readme and to references/make_index.py (if appropriate).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -44,8 +39,8 @@
     'keywords': 'radio astronomy interferometry',
     'extras_require': {
         'sim': ['mpi4py>=3.0.0', 'psutil'],
-        'all': ['mpi4py>=3.0.0', 'psutil', 'line_profiler', 'h5py'],
```
Comment (steven-murray) [inline setup.py:47]: Do we not need h5py anymore at all?

Comment (bhazelton) [reply setup.py:47]: We do, but we need it always so I moved it to the `install_requires` line

**Comment by bhazelton** [comment]
@aelanman are you ok with me merging this?

**Comment by aelanman** [comment]
@bhazelton That's fine. I can rebase etc. (Your branch on pyradiosky should be ready too, by the way, but you should probably check my changes to the circumpolar_nonrising test.)

**Pull request merged**